lib/dal/src/pkg/import.rs

```diff
@@ -94,38 +92,65 @@ async fn import_change_set(
     Vec<bool /*ImportEdgeSkip*/>,
 )> {
     // let default_change_set_id = ctx.get_workspace_default_change_set_id().await?;
+
+    // Cache the intrinsic funcs pkg in case we need it.
+    let unsafe_to_install_intrinsic_funcs_pkg = SiPkg::load_from_spec(IntrinsicFunc::pkg_spec()?)?;
+
     for func_spec in funcs {
-        let unique_id = func_spec.unique_id().to_string();
+        if let Some(intrinsic) = IntrinsicFunc::maybe_from_str(func_spec.name()) {
+            let maybe_func_id = match intrinsic {
+                IntrinsicFunc::ResourcePayloadToValue | IntrinsicFunc::NormalizeToArray => {
+                    Func::find_id_by_name_and_kind(ctx, func_spec.name(), FuncKind::Intrinsic)
+                        .await?
+                }
+                _ => Func::find_id_by_name(ctx, func_spec.name()).await?,
+            };
 
-        // This is a hack because the hash of the intrinsics has changed from the version in the
-        // packages. We also apply this to si:resourcePayloadToValue since it should be an
-        // intrinsic but is only in our packages
-        if func::is_intrinsic(func_spec.name()) || SPECIAL_CASE_FUNCS.contains(&func_spec.name()) {
-            if let Some(func_id) = Func::find_id_by_name(ctx, func_spec.name()).await? {
+            if let Some(func_id) = maybe_func_id {
                 let func = Func::get_by_id_or_error(ctx, func_id).await?;
 
-                thing_map.insert(unique_id.to_owned(), Thing::Func(func.to_owned()));
+                thing_map.insert(
+                    func_spec.unique_id().to_owned(),
+                    Thing::Func(func.to_owned()),
+                );
             } else {
+                let mut override_intrinsic_func_specs =
+                    unsafe_to_install_intrinsic_funcs_pkg.funcs_for_name(intrinsic.name())?;
+                let override_intrinsic_func_spec = override_intrinsic_func_specs.pop().ok_or(
+                    PkgError::IntrinsicFuncSpecsNoneForName(intrinsic.name().to_owned()),
+                )?;
+                if !override_intrinsic_func_specs.is_empty() {
+                    return Err(PkgError::IntrinsicFuncSpecsMultipleForName(
+                        intrinsic.name().to_owned(),
+                    ));
+                }
+
+                // We need to override the unique ID so that accessors grab the correct func.
+                thing_map.insert_override(
+                    func_spec.unique_id().into(),
+                    override_intrinsic_func_spec.unique_id().into(),
+                );
+
                 let func = import_func(
                     ctx,
-                    func_spec,
+                    &override_intrinsic_func_spec,
                     installed_module.clone(),
                     thing_map,
-                    options.create_unlocked,
+                    false,
```

Comment:
We learned that you may not need the overrides map at all anymore so long as the `func_spec.unique_id()` is pointing to the right "thing". However, this import code used for `import_change_set` calls `import_func` which handles thing map insertion internally... so you need the override unless you change `import_func` to have a new parameter to say "hey, use this override key to insert into the thing map instead". If you're reading this and your brain is fried, mine is too.